### ISSUE TITLE
如何使用您的损失函数？

### ISSUE THREAD

**Issue by huguangcheng** [xqwang14/SMS-Loss#1]
您好，我想试试您的损失函数，加入到我的模型训练过程中，看看效果，请问我该如何做?蟹蟹您的帮助！

**Comment by xqwang14** [comment]
Thanks for your interest to our work.

If you just want to finetune the model, just download pretrain models from avion and run "bash mir_finetune_vitb.sh". Remember change the path to your dataset first.

If you want to use SMS loss in different tasks, please first check if there is a relevancy matrix (a soft label set) for your task.

If so, you can find the function named "SymmetricMultiSimiliarityLoss" in [losses.py.](https://github.com/xqwang14/SMS-Loss/tree/main/avion/losses/losses.py)

Compared to other losses like max margin loss (hinge loss) that requires three parameters, image features, text features, and labels (namely "weight" as it is a soft label set in our task), we require an additional parameter called relevancy_weight. It is in fact a collection of positions (in the form of [y, x]) of soft labels in the relevancy matrix.

In our [dataloader](https://github.com/xqwang14/SMS-Loss/tree/main/avion/data/clip_dataset.py), we have a function called "VideoCaptionDatasetCLIP", which could return two parameters called 'i' and 'pos'. 'i' refers to the row of the corresponding soft label, and 'pos' refers to the colomn.

Line 460-462 of our [finetune script](https://github.com/xqwang14/SMS-Loss/tree/main/scripts/ammplus_finetune.py) shows how to assemble the parameter "relevancy_weight" in our SMS loss:


        relevancy_col = inputs.pop().unsqueeze(0) # pop 'pos' to relevancy_col
        relevancy_row = inputs.pop().unsqueeze(0) # pop 'i' to relevancy_row
        relevancy_weight = torch.cat((relevancy_col, relevancy_row), 0)

Our SMS loss will rebuild the relevancy matrix according to the collection of positions during forward process. 

We will provide a detailed instruction soon.

**Comment by huguangcheng** [comment]
好的，非常感谢您！
